modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/modal_content/SelectionFilter.tsx

```diff
@@ -77,179 +84,269 @@ function Body({
-									event.target.value
-							)
-						);
+			{Liferay.FeatureFlags['LPD-10754'] && (
+				<>
+					<ClayLayout.SheetSection className="mb-4">
+						<h3 className="sheet-subtitle">
+							{Liferay.Language.get('filter-source')}
+						</h3>
 
-						onPreselectedValuesChange([]);
-					}}
-					options={[
-						{
-							disabled: true,
-							label: Liferay.Language.get('select'),
-							value: '',
-						},
-						...picklists.map((item) => ({
-							label: item.name,
-							value: item.externalReferenceCode,
-						})),
-					]}
-					title={Liferay.Language.get('source-options')}
-					value={selectedPicklist?.externalReferenceCode || ''}
-				/>
-			</ClayForm.Group>
+						<ClayForm.Text>
+							{Liferay.Language.get(
+								'filter-source-determines-the-values-to-be-offered-in-this-filter-to-the-user'
+							)}
+						</ClayForm.Text>
+					</ClayLayout.SheetSection>
 
-			{selectedPicklist && (
-				<>
 					<ClayForm.Group>
-						<label htmlFor={multipleFormElementId}>
-							{Liferay.Language.get('selection')}
+						<label htmlFor={sourceOptionFormElementId}>
+							{Liferay.Language.get('source')}
 
-							<span
-								className="label-icon lfr-portal-tooltip ml-2"
-								title={Liferay.Language.get(
-									'determines-how-many-preselected-values-for-the-filter-can-be-added'
-								)}
-							>
-								<ClayIcon symbol="question-circle-full" />
-							</span>
+							<RequiredMark />
 						</label>
 
-						<ClayRadioGroup
-							name={multipleFormElementId}
-							onChange={(newVal: any) => {
-								onMultipleChange(newVal === 'true');
+						<ClaySelectWithOption
+							aria-label={Liferay.Language.get(
+								'choose-an-option'
+							)}
+							name={sourceOptionFormElementId}
+							onChange={(event) => {
+								onSourceChange(
+									event.target
+										.value as ESelectionFilterSourceType
+								);
 							}}
-							value={multiple ? 'true' : 'false'}
-						>
-							<ClayRadio
-								label={Liferay.Language.get('multiple')}
-								value="true"
-							/>
-
-							<ClayRadio
-								label={Liferay.Language.get('single')}
-								value="false"
-							/>
-						</ClayRadioGroup>
+							options={[
+								{
+									disabled: true,
+									label: Liferay.Language.get(
+										'choose-an-option'
+									),
+									value: '',
+								},
+								{
+									disabled: false,
+									label: Liferay.Language.get(
+										'object-picklist'
+									),
+									value: ESelectionFilterSourceType.PICKLIST,
+								},
+							]}
+							required
+							title={Liferay.Language.get('source')}
+							value={sourceType || ''}
+						/>
 					</ClayForm.Group>
-					<ClayForm.Group
-						className={classNames({
-							'has-error': !isValidSingleMode,
-						})}
-					>
-						<label htmlFor={preselectedValuesFormElementId}>
-							{Liferay.Language.get('preselected-values')}
+				</>
+			)}
+
+			{Liferay.FeatureFlags['LPD-10754'] ? (
```

Comment:
@markocikos This is the breakpoint between the old UI and the new one (that will select which component should be rendered based on source type - picklists, api, etc), currently we only have ObjectPicklist.tsx until the next story